maas/resource_maas_raid_test.go

```diff
@@ -0,0 +1,510 @@
+
+					resource.TestCheckResourceAttr("maas_raid.test", "block_devices.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "partitions.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "spare_devices.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "spare_partitions.#", "1"),
+
+					resource.TestCheckTypeSetElemAttrPair("maas_raid.test", "block_devices.0", fmt.Sprintf("maas_block_device.%v", blockDevice1Name), "id"),
+					resource.TestCheckResourceAttrPair("maas_raid.test", "partitions.0", fmt.Sprintf("maas_block_device.%v", blockDevice4Name), "partitions.0.id"),
+					resource.TestCheckTypeSetElemAttrPair("maas_raid.test", "spare_devices.0", fmt.Sprintf("maas_block_device.%v", blockDevice3Name), "id"),
+					resource.TestCheckResourceAttrPair("maas_raid.test", "spare_partitions.0", fmt.Sprintf("maas_block_device.%v", blockDevice2Name), "partitions.0.id"),
+				),
+			},
+			// Test the worst-case operation to ensure update is working correctly: fully swapping active and spare disks
+			{
+				Config: baseConfig +
+					testAccRAIDConfig(swappedName, level, swappedFsType, swappedMountPoint,
+						generateRAIDBlockDevices([]string{blockDevice3Name}),
+						generateRAIDPartitions([]string{blockDevice2Name}),
+						generateRAIDBlockDevices([]string{blockDevice1Name}),
+						generateRAIDPartitions([]string{blockDevice4Name}),
+					),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRAIDExists("maas_raid.test"),
+					resource.TestCheckResourceAttr("maas_raid.test", "name", swappedName),
+					resource.TestCheckResourceAttr("maas_raid.test", "level", level),
+					resource.TestCheckResourceAttr("maas_raid.test", "fs_type", swappedFsType),
+					resource.TestCheckResourceAttr("maas_raid.test", "mount_point", swappedMountPoint),
+
+					resource.TestCheckResourceAttr("maas_raid.test", "block_devices.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "partitions.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "spare_devices.#", "1"),
+					resource.TestCheckResourceAttr("maas_raid.test", "spare_partitions.#", "1"),
+
+					resource.TestCheckTypeSetElemAttrPair("maas_raid.test", "block_devices.0", fmt.Sprintf("maas_block_device.%v", blockDevice3Name), "id"),
+					resource.TestCheckResourceAttrPair("maas_raid.test", "partitions.0", fmt.Sprintf("maas_block_device.%v", blockDevice2Name), "partitions.0.id"),
+					resource.TestCheckTypeSetElemAttrPair("maas_raid.test", "spare_devices.0", fmt.Sprintf("maas_block_device.%v", blockDevice1Name), "id"),
+					resource.TestCheckResourceAttrPair("maas_raid.test", "spare_partitions.0", fmt.Sprintf("maas_block_device.%v", blockDevice4Name), "partitions.0.id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccResourceMAASRAID_differentLevels(t *testing.T) {
+	// We need to test a RAID can be created for every level supported
+	// TODO: Update this when LP:2109708 is released in MAAS
```

Comment:
Please do not forget this after resolving the suggestion for how to treat raid 10